lib/src/layer/tile_layer/tile_image_manager.dart

```diff
@@ -125,10 +125,16 @@ class TileImageManager {
     final tilesToReload = List<TileImage>.from(_tiles.values);
 
     for (final tile in tilesToReload) {
-      tile.imageProvider = layer.tileProvider.getImage(
-        tileBounds.atZoom(tile.coordinates.z).wrap(tile.coordinates),
-        layer,
-      );
+      tile.imageProvider = layer.tileProvider.supportsCancelLoading
+          ? layer.tileProvider.getImageWithCancelLoadingSupport(
+              tileBounds.atZoom(tile.coordinates.z).wrap(tile.coordinates),
+              layer,
+              tile.cancelLoading.future,
+            )
+          : layer.tileProvider.getImage(
+              tileBounds.atZoom(tile.coordinates.z).wrap(tile.coordinates),
+              layer,
+            );
```

Comment:
Why is this logic not inside the TileProvider if the tile manager asks the TileProvider what function to call anyways.

Reply:
See below, in summary, to reduce breaking chnages.